infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/RestApiRequest.java

```diff
@@ -54,6 +54,10 @@ public interface RestApiRequest {
 
   <T> List<T> getQueryParameterList(final ParameterMetadata<T> parameterMetadata);
 
+  <T> Optional<T> getOptionalHeader(final ParameterMetadata<T> parameterMetadata);
+
+  <T> T getHeader(final ParameterMetadata<T> parameterMetadata);
```

Comment:
NIT: If we want to differentiate from the headers we add to the response, we could name these methods `getRequestHeader` and `getOptionalRequestHeader`.